Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1

```diff
@@ -0,0 +1,356 @@
+    Write-Host "Required modules are: $($requiredModule -join ', ')" -ForegroundColor Cyan
+    Write-Host "Checking for required modules..."
+    $availableModules = $(Get-Module -ListAvailable $requiredModule)
+
+    # Checking if required module is available in user machine
+    $requiredModule | ForEach-Object {
+        if($availableModules.Name -notcontains $_)
+        {
+            Write-Host "Installing module $($_)..." -ForegroundColor Yellow
+            Install-Module -Name $_ -Scope CurrentUser -Repository 'PSGallery'
+        }
+        else {
+            Write-Host "$($_) module is available." -ForegroundColor Green
+        }
+    }
+}
+
+function Disable-RemoteDesktopAccess
+{
+    <#
+        .SYNOPSIS
+        This command would help in remediating 'Azure_CloudService_SI_Disable_RemoteDesktop_Access' control.
+        .DESCRIPTION
+        This command would help in remediating 'Azure_CloudService_SI_Disable_RemoteDesktop_Access' control.
+        .PARAMETER SubscriptionId
+            Enter subscription id on which remediation need to perform.
+        .PARAMETER Path
+            Json file path which contain failed controls detail to remediate.
+        .Parameter Force
+            To disable RDP access forcefully.
+    #>
+
+    param (
+        [string]
+        [Parameter(Mandatory = $true, HelpMessage="Enter subscription id for remediation")]
+        $SubscriptionId,
+
+        [string]
+        [Parameter(Mandatory = $false, HelpMessage="Json file path which contain cloud service(s) details to remediate")]
+        $Path,
+
+        [switch]
+        $Force
+    )
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "Starting to disable remote desktop access on cloud service(s) from subscription [$($SubscriptionId)]..."
+    Write-Host $([Constants]::SingleDashLine)
+    
+    try 
+    {
+        Write-Host "Checking for pre-requisites..."
+        Pre_requisites
+        Write-Host $([Constants]::SingleDashLine)     
+    }
+    catch 
+    {
+        Write-Host "Error occured while checking pre-requisites. ErrorMessage [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)    
+        break
+    }
+    
+    # Connect to AzAccount
+    $isContextSet = Get-AzContext
+    if ([string]::IsNullOrEmpty($isContextSet))
+    {       
+        Write-Host "Connecting to AzAccount..."
+        Connect-AzAccount -ErrorAction Stop
+        Write-Host "Connected to AzAccount" -ForegroundColor $([Constants]::MessageType.Update)
+    }
+
+    # Setting context for current subscription.
+    $currentSub = Set-AzContext -SubscriptionId $SubscriptionId -Force -ErrorAction Stop
+    Select-AzureSubscription -SubscriptionId $($SubscriptionId) -ErrorAction Stop
+    
+    Write-Host "Note: `n Cloud services on which RDP was enabled, during the deployment will not be remediated." -ForegroundColor Yellow
+    Write-Host "------------------------------------------------------"
```

Comment:
Is rollback supported? If not yet, and if rollback is not possible, please include a prompt that asks for the user's input before proceeding.

Reply:
Added